cpp/src/arrow/compute/row/compare_internal_avx2.cc

```diff
@@ -251,6 +253,35 @@ uint32_t KeyCompare::CompareBinaryColumnToRowHelper_avx2(
   }
 }
 
+namespace {
+
+/// Intrinsics `_mm256_i32gather_epi32/64` treat the `vindex` as signed integer, and we
```

Comment:
Can you use regular comments (`//`)? This isn't a docstring so shouldn't use the docstring-specific prefix (`///`)

Reply:
Yeah, will do.

Reply:
Addressed.